parlai/core/teachers.py

```diff
@@ -448,10 +448,20 @@ def observe(self, observation):
         """
         Process observation for metrics.
         """
+        self.metrics.clear_recent()
         if hasattr(self, 'lastY') and self.lastY is not None:
             self.metrics.evaluate_response(observation, self.lastY)
             self.custom_evaluation(self.last_act, self.lastY, observation)
             self.lastY = None
+        recent_metrics = self.metrics.report_recent()
+        if recent_metrics:
+            # for display purposes (display_model), take all accumulated
+            # metrics back into the original observation. This is an abuse of
```

Comment:
If it is an abuse, is there a better way of doing it later to add as a TODO?

Reply:
I'm not aware of one, otherwise I would have just done it.